src/timeunit.ts

```diff
@@ -3,38 +3,69 @@ import {ScaleType} from './scale';
 import {Dict, keys} from './util';
 import * as log from './log';
 
-export enum TimeUnit {
-  YEAR = 'year' as any,
-  MONTH = 'month' as any,
-  DAY = 'day' as any,
-  DATE = 'date' as any,
-  HOURS = 'hours' as any,
-  MINUTES = 'minutes' as any,
-  SECONDS = 'seconds' as any,
-  MILLISECONDS = 'milliseconds' as any,
-  YEARMONTH = 'yearmonth' as any,
-  YEARMONTHDATE = 'yearmonthdate' as any,
-  YEARMONTHDATEHOURS = 'yearmonthdatehours' as any,
-  YEARMONTHDATEHOURSMINUTES = 'yearmonthdatehoursminutes' as any,
-  YEARMONTHDATEHOURSMINUTESSECONDS = 'yearmonthdatehoursminutesseconds' as any,
-
-  // MONTHDATE always include 29 February since we use year 0th (which is a leap year)
-  MONTHDATE = 'monthdate' as any,
-  HOURSMINUTES = 'hoursminutes' as any,
-  HOURSMINUTESSECONDS = 'hoursminutesseconds' as any,
-  MINUTESSECONDS = 'minutesseconds' as any,
-  SECONDSMILLISECONDS = 'secondsmilliseconds' as any,
-  QUARTER = 'quarter' as any,
-  YEARQUARTER = 'yearquarter' as any,
-  QUARTERMONTH = 'quartermonth' as any,
-  YEARQUARTERMONTH = 'yearquartermonth' as any,
+export namespace TimeUnit {
+  export const YEAR: 'year' = 'year';
+  export type YEAR = typeof YEAR;
+  export const MONTH: 'month' = 'month';
+  export type MONTH = typeof MONTH;
+  export const WEEK: 'week' =  'week';
+  export type WEEK = typeof WEEK;
```

Comment:
{This line here](https://github.com/vega/vega-lite/blob/master/test/compile/timeunit.test.ts#L44) is the test wrong? It explicitly calls out using `'week'`.